exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/Experience.java

```diff
@@ -26,13 +26,13 @@
 
     public ExpObservable<Experience> getCurrentExperience(){
         final ExpObservable<Device> observable = Device.getCurrentDevice();
-            return new ExpObservable<Experience>(observable.<Experience>flatMap(new Func1<Device, Observable<Experience>>() {
+            return new ExpObservable<Experience>(observable.<Experience>flatMapExp(new Func1<Device, ExpObservable<Experience>>() {
                 @Override
-                public Observable<Experience> call(Device device) {
+                public ExpObservable<Experience> call(Device device) {
                     if(device!=null && device.getExperience()!=null){
```

Comment:
won't calling getExperience() for the null check call the api twice?  this should probably check the document like device.get('experience.uuid') or something similar right?